File: output/deepseek_r1/avatar/algo_based_translation/python/java/codeforces_518_B.java

import java.util.HashMap;
import java.util.Map;
import java.util.Scanner;

public class codeforces_518_B {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        String s = scanner.nextLine();
        String t = scanner.nextLine();
        scanner.close();

        Map<Character, Integer> freqs = new HashMap<>();
        for (char c : s.toCharArray()) {
            freqs.put(c, freqs.getOrDefault(c, 0) + 1);
        }

        Map<Character, Integer> freqt = new HashMap<>();
        for (char c : t.toCharArray()) {
            freqt.put(c, freqt.getOrDefault(c, 0) + 1);
        }

        int yay = 0;
        for (Character c : freqs.keySet()) {
            while (freqs.get(c) > 0 && freqt.containsKey(c) && freqt.get(c) > 0) {
                yay++;
                freqs.put(c, freqs.get(c) - 1);
                freqt.put(c, freqt.get(c) - 1);
            }
        }

        int whoops = 0;
        for (Character c : freqs.keySet()) {
            while (freqs.get(c) > 0) {
                if (Character.isLowerCase(c)) {
                    char upper = Character.toUpperCase(c);
                    if (freqt.containsKey(upper) && freqt.get(upper) > 0) {
                        whoops++;
                        freqs.put(c, freqs.get(c) - 1);
                        freqt.put(upper, freqt.get(upper) - 1);
                    } else {
                        break;
                    }
                } else if (Character.isUpperCase(c)) {
                    char lower = Character.toLowerCase(c);
                    if (freqt.containsKey(lower) && freqt.get(lower) > 0) {
                        whoops++;
                        freqs.put(c, freqs.get(c) - 1);
                        freqt.put(lower, freqt.get(lower) - 1);
                    } else {
                        break;
                    }
                } else {
                    break;
                }
            }
        }

        System.out.println(yay + " " + whoops);
    }
}

File: dataset/avatar/Python/Code/codeforces_518_B.py
s , t = input ( ) , input ( )
freqs = dict ( )
freqt = dict ( )
yay , whoops = 0 , 0
for letra in s :
    if letra in freqs :
        freqs [ letra ] += 1
    else :
        freqs [ letra ] = 1
for letra in t :
    if letra in freqt :
        freqt [ letra ] += 1
    else :
        freqt [ letra ] = 1
for letra in freqs.keys ( ) :
    while ( freqs [ letra ] > 0 and letra in freqt and freqt [ letra ] > 0 ) :
        yay += 1
        freqs [ letra ] -= 1
        freqt [ letra ] -= 1
for letra in freqs.keys ( ) :
    while ( freqs [ letra ] > 0 ) :
        if letra.islower ( ) and letra.upper ( ) in freqt.keys ( ) and freqt [ letra.upper ( ) ] > 0 :
            whoops += 1
            freqs [ letra ] -= 1
            freqt [ letra.upper ( ) ] -= 1
        elif letra.isupper ( ) and letra.lower ( ) in freqt.keys ( ) and freqt [ letra.lower ( ) ] > 0 :
            whoops += 1
            freqs [ letra ] -= 1
            freqt [ letra.lower ( ) ] -= 1
        else :
            break ;
print ( yay , whoops )

